### ISSUE TITLE
Tags of previous revisions getting removed

### ISSUE THREAD

**Issue by sra-013** [GoogleCloudPlatform/gcr-cleaner#143]
### TL;DR

Running the clear on a repo is removing tags for the previous revision.

### Expected behavior

The tags should not get affected.

### Observed behavior

Tags are getting removed.

### How are you running gcr-cleaner?

Server

### gcr-cleaner version

latest

### Environment

have added the following step to my inline cloud build yaml file (with the correct project, repo and image names substituted)

```
name: 'asia-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner-cli:latest'
    args:
      - '-repo'
      - asia.gcr.io/my-project/my-repo/my-image
      - '-keep'
      - '10'
      - '-tag-filter-any'
      - .*
```

### Additional information

The correct number of images are being preserved, the rest are getting deleted. The only issue is tags getting deleted for the previous revision.

**Comment by sethvargo** [comment]
Hi @sra-013 - please provide the [debug logs](https://github.com/GoogleCloudPlatform/gcr-cleaner#debugging) including the pre-run and post-run state.

**Comment by sra-013** [comment]
Hi @sethvargo, have attached pictures of the pre and post run states.

<img width="1211" alt="Screenshot 2023-04-20 at 6 30 51 PM" src="https://user-images.githubusercontent.com/14370496/233623025-4f47be7a-2a93-49c3-8798-568d88e3475b.png">
<img width="1188" alt="Screenshot 2023-04-20 at 6 35 33 PM" src="https://user-images.githubusercontent.com/14370496/233623055-11bae34c-b993-47eb-a725-a9db4b0191ff.png">

**Comment by sethvargo** [comment]
Hi @sra-013 - sorry, I'm not sure what I'm looking at here. Are these multi-arch images?

**Comment by sra-013** [comment]
Hey @sethvargo - these are images created by builds triggered on github merges to main. I also noticed that the bug seems to happen only on manually triggering a build, but i am yet to confirm this.

**Comment by sra-013** [comment]
Also, does this not clean images from cloud storage?

**Comment by sethvargo** [comment]
What do you mean "from Cloud Storage"?

**Comment by sra-013** [comment]
GCR stores container images in Google Cloud Storage buckets per my understanding. Please correct me if I am wrong.

**Comment by sra-013** [comment]
Yep we are using Container Registry and the images did get deleted but the cloud storage buckets still seem quite full. Would I maybe need to set a flag in my build step for cloud storage to get cleaned up as well?

**Comment by sethvargo** [comment]
What do you mean by "quite full"? What is in the Cloud Storage bucket?

**Comment by sra-013** [comment]
I have a total of 30 images when i look at gcr but the corresponding gcs bucket (asia.artifacts.instasell-live.appspot.com/containers/images) has 450 objects.

**Comment by sethvargo** [comment]
Deleting images from Container Registry will delete them from Cloud Storage.
